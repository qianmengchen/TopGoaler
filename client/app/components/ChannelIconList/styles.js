import { StyleSheet } from 'react-native';

const list = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-around'
  },
  iconContainer: {
    justifyContent: 'center'
  },
  icon: {
    height: 80,
    width: 80,
    marginHorizontal: 10,
    borderWidth: 1,
    borderRadius: 40
  }
});

export { list };
