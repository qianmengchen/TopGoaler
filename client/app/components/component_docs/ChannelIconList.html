<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: ChannelIconList</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: ChannelIconList</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>ChannelIconList<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="ChannelIconList"><span class="type-signature"></span>new ChannelIconList<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>navigate</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last">A function that allows the user to nagivate to another component.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>channels</code></td>
            

            <td class="type">
            
                
<span class="param-type">Object</span>


            
            </td>

            

            

            <td class="description last">An object that contains all channels.</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="ChannelIconList.js.html">ChannelIconList.js</a>, <a href="ChannelIconList.js.html#line12">line 12</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="ChannelIconList.html">ChannelIconList</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Sun Dec 02 2018 21:00:24 GMT-0800 (Pacific Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>