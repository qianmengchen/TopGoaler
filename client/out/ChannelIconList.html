<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: ChannelIconList</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: ChannelIconList</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>ChannelIconList<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="ChannelIconList"><span class="type-signature"></span>new ChannelIconList<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="ChannelIconList.js.html">ChannelIconList.js</a>, <a href="ChannelIconList.js.html#line11">line 11</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="ChannelIconList.html">ChannelIconList</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Sun Dec 02 2018 20:38:44 GMT-0800 (Pacific Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>