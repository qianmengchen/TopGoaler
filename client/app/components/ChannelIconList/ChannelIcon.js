import React from 'react';
import { Image, TouchableOpacity } from 'react-native';
import * as styles from './styles';

/**
 * @class ChannelIcon
 * @property {Object} item - An object describing a task.
 * @property {function} goToChannel - A function that allows the user to navigate to the specified channel.
 */
const ChannelIcon = props => {
  const { item, goToChannel } = props;

  return (
    <TouchableOpacity
      style={styles.list.iconContainer}
      onPress={() => goToChannel(item.id)}
    >
      <Image style={styles.list.icon} source={{ uri: item.image_url }} />
    </TouchableOpacity>
  );
};

export default ChannelIcon;
