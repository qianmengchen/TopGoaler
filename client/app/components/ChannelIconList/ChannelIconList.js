import React, { Component } from 'react';
import { View, FlatList } from 'react-native';
import * as styles from './styles';
import ChannelIcon from './ChannelIcon';

class ChannelIconList extends Component {
  constructor(props) {
    super(props);

    this.fakeProps = {
      channels: [
        {
          name: 'leetcoder',
          icon: require('../../images/leetcode.jpg')
        },
        {
          name: 'runner',
          icon: require('../../images/run.jpg')
        },
        {
          name: 'GRE',
          icon: require('../../images/GRE.jpg')
        },
        {
          name: 'GRE1',
          icon: require('../../images/GRE.jpg')
        },
        {
          name: 'GRE2',
          icon: require('../../images/GRE.jpg')
        },
        {
          name: 'GRE3',
          icon: require('../../images/GRE.jpg')
        }
      ]
    };
  }

  render() {
    return (
      <View style={styles.list.container}>
        <FlatList
          horizontal
          data={this.fakeProps.channels}
          renderItem={ChannelIcon}
          keyExtractor={item => item.name}
        />
      </View>
    );
  }
}

export default ChannelIconList;
