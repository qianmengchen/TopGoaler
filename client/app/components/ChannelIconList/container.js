import { connect } from 'react-redux';
import ChannelIconList from './ChannelIconList';

const mapStateToProps = state => ({
  state: state
});

export default connect(mapStateToProps)(ChannelIconList);
