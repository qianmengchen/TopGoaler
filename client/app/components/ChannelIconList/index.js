import ChannelIconList from './container';

export { ChannelIconList };
