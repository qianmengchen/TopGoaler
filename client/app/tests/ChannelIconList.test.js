import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
require('isomorphic-fetch');

import ChannelIconList from '../components/ChannelIconList/ChannelIconList';

const middlewares = [thunkMiddleware]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialTaskState = {
  tasks: ['do frontend', 'do backend', 'do MVP ftw'],
  isFetching: false,
  currentTaskPage: 'ChannelListPage'
};

describe('Testing Channel Icon List component', () => {
  const wrapper = shallow(<ChannelIconList />, {
    context: { store: mockStore(initialTaskState) }
  });
  const render = wrapper.dive();

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });

  it('should have 1 FlatList for icons', () => {
    expect(render.find('FlatList')).toHaveLength(1);
  });
});
