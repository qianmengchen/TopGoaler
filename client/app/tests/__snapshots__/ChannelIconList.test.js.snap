// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Channel Icon List component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <FlatList
      disableVirtualization={false}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FlatList
        disableVirtualization={false}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />,
      "style": Object {
        "flex": 1,
        "justifyContent": "space-around",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data": undefined,
        "disableVirtualization": false,
        "horizontal": true,
        "initialNumToRender": 10,
        "keyExtractor": [Function],
        "maxToRenderPerBatch": 10,
        "numColumns": 1,
        "onEndReachedThreshold": 2,
        "renderItem": [Function],
        "scrollEventThrottle": 50,
        "updateCellsBatchingPeriod": 50,
        "windowSize": 21,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FlatList
          disableVirtualization={false}
          horizontal={true}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />,
        "style": Object {
          "flex": 1,
          "justifyContent": "space-around",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": undefined,
          "disableVirtualization": false,
          "horizontal": true,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
